oxide/lib.go

```diff
@@ -118,6 +158,89 @@ func defaultUserAgent() string {
 	return fmt.Sprintf("oxide.go/%s", version)
 }
 
+// getProfile determines the path of the user's credentials file
+// and returns the host and token for the requested profile.
+func getProfile(cfg Config) (*authCredentials, error) {
+	if cfg.Host != "" || cfg.Token != "" {
+		return nil, errors.New("cannot authenticate with both a profile and host/token")
+	}
+
+	configDir := cfg.ConfigDir
+	if configDir == "" {
+		homeDir, err := os.UserHomeDir()
+		if err != nil {
+			return nil, fmt.Errorf("unable to find user's home directory: %w", err)
+		}
+		configDir = filepath.Join(homeDir, defaultConfigDir)
+	}
+
+	profile := cfg.Profile
+
+	// Use explicitly configured profile over default when both are set.
+	if cfg.UseDefaultProfile && profile == "" {
+		configPath := filepath.Join(configDir, configFile)
+
+		var err error
+		profile, err = defaultProfile(configPath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get default profile from %q: %w", configPath, err)
+		}
+	}
+
+	credentialsPath := filepath.Join(configDir, credentialsFile)
+	fileCreds, err := parseCredentialsFile(credentialsPath, profile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get credentials for profile %q from %q: %w", profile, credentialsPath, err)
+	}
+
+	return fileCreds, nil
+}
+
+// defaultProfile returns the default profile from config.toml, if present.
+func defaultProfile(configPath string) (string, error) {
+	configFile, err := toml.LoadFile(configPath)
+	if err != nil {
+		return "", fmt.Errorf("failed to open config: %w", err)
```

Comment:
Argh, sorry missed this one! Can this error say what configuration file failed to open? Something like "failed to open configuration file '.config/oxide/credentials.toml': %w, err "

Reply:
We're capturing the path in the error one level up. The [`fails with invalid config dir` test output](https://github.com/oxidecomputer/oxide.go/blob/3d0217bebd5db2bff5e84dbb47ac604b8742195e/oxide/lib_test.go#L367) shows the formatting.